spring-integration-core/src/test/java/org/springframework/integration/channel/interceptor/ObservationPropagationChannelInterceptorTests.java

```diff
@@ -199,49 +207,55 @@ void observationPropagatedOverQueueChannel() throws InterruptedException {
-					.observe(() -> {
-						// Let's assume that this is the user code
-					});
-			handleLatch.countDown();
-		});
+	void observationContextPropagatedOverExecutorChannel() {
+		BridgeHandler handler = new BridgeHandler();
+		handler.registerObservationRegistry(this.observationRegistry);
+		handler.setBeanName("testBridge");
+		this.testTracingChannel.subscribe(handler);
 
-		// This would be the instrumentation code on the sender side (user's code would call e.g. MessageTemplate and this code
-		// would lay in MessageTemplate)
-		// We need to mutate the carrier, so we need to use the builder not the message since message headers are immutable
-		SenderContext<MessageBuilder<String>> senderContext =
-				new SenderContext<>((carrier, key, value) -> Objects.requireNonNull(carrier).setHeader(key, value));
-		MessageBuilder<String> builder = MessageBuilder.withPayload("test");
-		senderContext.setCarrier(builder);
+		QueueChannel replyChannel = new QueueChannel();
 
-		Observation.createNotStarted("sending", senderContext, this.observationRegistry)
-				.observe(() -> this.testTracingChannel.send(builder.build()));
+		MutableMessage<String> message =
+				(MutableMessage<String>) MutableMessageBuilder.withPayload("test")
+						.setHeader(MessageHeaders.REPLY_CHANNEL, replyChannel)
+						.build();
 
-		assertThat(handleLatch.await(10, TimeUnit.SECONDS)).isTrue();
+		Observation.createNotStarted("sending", new MessageSenderContext(message), this.observationRegistry)
+				.observe(() -> this.testTracingChannel.send(message));
+
+		Message<?> receive = replyChannel.receive();
+
+		assertThat(receive).isNotNull()
+				.extracting(Message::getHeaders)
+				.asInstanceOf(InstanceOfAssertFactories.MAP)
+				.containsEntry("foo", "some foo value")
+				.containsEntry("bar", "some bar value");
 
 		TestObservationRegistryAssert.assertThat(this.observationRegistry)
 				.doesNotHaveAnyRemainingCurrentObservation();
 
 		TracerAssert.assertThat(this.simpleTracer)
 				.reportedSpans()
 				.hasSize(2)
-				.satisfies(simpleSpans -> SpansAssert.assertThat((Collection<FinishedSpan>) (Collection) simpleSpans)
+				.satisfies(simpleSpans -> assertSpans(simpleSpans)
 						.hasASpanWithName("sending")
-						.assertThatASpanWithNameEqualTo("user.code")
+						.assertThatASpanWithNameEqualTo("testBridge receive")
 						.hasTag("foo", "some foo value")
 						.hasTag("bar", "some bar value")
+						.hasTag("spring.integration.type", "handler")
+						.hasTag("spring.integration.name", "testBridge")
 						.hasKindEqualTo(Span.Kind.CONSUMER));
+
+		assertThat(this.meterRegistry.get("spring.integration.handler")
+				.tag("spring.integration.name", "testBridge")
+				.tag("spring.integration.type", "handler")
+				.tag("error", "none")
+				.timer().count()).isEqualTo(1);
+	}
+
+	@SuppressWarnings("unchecked")
+	private static SpansAssert assertSpans(Collection<? extends FinishedSpan> actual) {
```

Comment:
Have you filed a PR / issue about this in micrometer-tracing?

Reply:
No, I did not yet.
Will look into that today if time permits.

Reply:
Here you are: https://github.com/micrometer-metrics/tracing/pull/65 !

Reply:
Merged, thanks!